### DISCUSSION TITLE
PiVPN hangs, possibly on heavy traffic

### DISCUSSION THREAD

**Discussion by anpap** [pivpn/pivpn#1461] [Support]
## In raising this issue, I confirm the following:

`{please fill the checkboxes, e.g: [X]}`

- [X] I have read the [documentation](https://docs.pivpn.io)
- [X] I have read and understood the [contributors guide](https://github.com/pivpn/pivpn/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [] The issue I am reporting can be *is* directly related to the pivpn installer script.

### Have you searched for similar issues and solutions?
    yes

### Has your install failed? 
```
no
```

#### What type of hardware are you running PiVPN at?

```

OrangePi
```
#### Output of `uname -a`
```
  Linux OrangePi 5.3.5+ #2 SMP Fri Nov 15 16:24:33 CST 2019 arm7l GNU/Linux
```

#### Output of `cat /etc/os-release`
```
  PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
VERSION_CODENAME=stretch
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

```

### If install failed Please provide the console output of  `curl -L https://install.pivpn.io | bash`
```
  OUTPUT HERE / ANSWER HERE
  DO NOT DELETE THE BACK-TICKS 
  PASTE THE OUTPUT INSIDE THE BACK-TICKS
```

### Issue

In some cases PiVPN hangs, possibly in cases where I try to use Remote Desktop (Microsoft RDP) or stream video from an IP security camera. 
Is there a way to troubleshoot and fix? 
Can you please advise?

### Console output of      `curl -L install.pivpn.io | bash`
```
Finds the existing installation and prompts for installation or repair. 
```

### Console output of      `pivpn add` or `pivpn add nopass`
```
  OUTPUT HERE
  DO NOT DELETE THE BACK-TICKS 
  PASTE THE OUTPUT INSIDE THE BACK-TICKS
```

### Console output of      `pivpn debug`

```
  OUTPUT HERE
  DO NOT DELETE THE BACK-TICKS 
  PASTE THE OUTPUT INSIDE THE BACK-TICKS
```
### Have you taken any steps towards solving your issue?
```
  which?
```

**Comment by orazioedoardo** [comment]
Not a PiVPN bug, but... Is your power supply adequate?

**Reply by anpap** [reply]
Seems to be adequate.... The issue occurs on two Orange Pis. The power supply for the main one is quite large...

**Comment by coolapso** [comment]
Moved to discussions.

**Comment by TinCanTech** [comment]
In my experience, MS RDP is deliberately broken when used as a free service. (Microsoft Windows Version dependent nobbling)

**Reply by anpap** [reply]
Thanks for the answers!
Haven't found a solution yet. As a workaround I created a scheduled task to reboot the OrangePIs every night. 
So far so good... :-)

**Reply by anpap** [reply]
Update:
Yesterday PiVPN hanged while using ZoneMinder over the VPN connection, to view live video from 2-3 IP cameras. 
Left it to see if the automatic reboot would bring it back online, but unfortunately the scheduled reboot task never ran. 
When it hangs it still responds to ping requests, accepts SSH connection but doesn't give a login prompt...

back to troubleshooting...

**Reply by orazioedoardo** [reply]
Check CPU usage with `htop`. Note that if you are using wireguard there is no process, just check the CPU bars.

**Reply by anpap** [reply]
Thanks for the suggestion, will do!
Does it have anything to do with using UDP vs TCP (as per default PiVPN configuration)?

**Reply by orazioedoardo** [reply]
I think TCP is more expensive on the CPU.
